andela/lotus-ah
# #160451609 Paginate articles and users route

#### What does this PR do?
    This PR enables pagination support on both users and articles route
#### Description of Task to be completed?
     1. Add pagination to users route 
     2. Add pagination to articles route
#### How should this be manually tested?
- Start the app locally on port `3000`
- Construct articles  url as follows ```/api/v1/articles/1```
- Make a `GET` request to the url
- You should get a response with the structure below:
```js
{
    "status": "SUCCESS",
    "message": "Fetched all article",
    "articles": [
        {
            "id": 2,
            "slug": "Policemen-of-this-days-are-not-yet-here-ce86c7f5-5423-4b3d-b9ac-df7cf2bd903a",
            "userId": 2,
            "title": "Policemen of this days are not yet here!",
            "body": "Doing good is right for you and I",
            "description": "Doing good is right but is generally \n        not common habit you can find on any one cute",
            "imageUrl": null,
            "rating": null,
            "createdAt": "2018-10-29T15:55:59.883Z",
            "updatedAt": "2018-10-29T15:55:59.883Z",
            "comments": [],
            "Tags": [],
            "reactions": [
                {
                    "id": 1,
                    "likes": false,
                    "dislike": false
                }
            ]
        },
        {
            "id": 4,
            "slug": "Policemen-of-this-days-are-not-yet-here-b8935f25-6c11-4054-b142-49db649f4b4d",
            "userId": 2,
            "title": "Policemen of this days are not yet here!",
            "body": "Doing good is right for you and I",
            "description": "Doing good is right but is generally \n        not common habit you can find on any one cute",
            "imageUrl": null,
            "rating": null,
            "createdAt": "2018-10-29T15:56:00.260Z",
            "updatedAt": "2018-10-29T15:56:00.260Z",
            "comments": [
                {
                    "id": 1,
                    "commentBody": "pholayemi is a boy",
                    "userId": 2,
                    "createdAt": "2018-10-29T15:56:00.292Z"
                }
            ],
            "Tags": [],
            "reactions": []
        },
        {
            "id": 3,
            "slug": "Policemen-of-this-days-are-not-yet-here-e348b7e1-366d-4927-ac9d-5ea742876283",
            "userId": 2,
            "title": "Policemen of this days are not yet here!",
            "body": "Doing good is right but is generally \n        not common habit you can find on any one cute",
            "description": "Doing good is right for you and I",
            "imageUrl": null,
            "rating": null,
            "createdAt": "2018-10-29T15:56:00.156Z",
            "updatedAt": "2018-10-29T15:56:00.156Z",
            "comments": [],
            "Tags": [],
            "reactions": []
        }
    ],
    "articlesCount": 9,
    "page": "1",
    "pages": 3
}
```
#### Any background context you want to provide?
      Pagination makes fetching articles more performant
#### What are the relevant pivotal tracker stories?
      #160451609
#### Screenshots
     N/A
#### Questions:
   None
